streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -364,11 +365,8 @@ private void commitState() {
             final long partitionTime = partitionTimes.get(partition);
             consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset, encodeTimestamp(partitionTime)));
         }
-        recordCollector.commit(consumedOffsetsAndMetadata);
 
-        commitNeeded = false;
-        commitRequested = false;
-        commitSensor.record(time.nanoseconds() - startNs);
+        return consumedOffsetsAndMetadata;
```

Comment:
Looks like we lack test coverage for TimeoutException and KafkaException cases

Reply:
Yeah, this PR does not yet add all required test...

Reply:
Covered via `shouldCommitNextOffsetFromQueueIfAvailable` and `shouldCommitConsumerPositionIfRecordQueueIsEmpty`